DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+Website](https://github.com/apache/daffodil-site).  When you are ready
+to contribute code from any of your forks to any of these upstream
+Apache repositories, you must create a pull request from your fork,
+allow the Apache Daffodil developers to review your contribution, make
+any suggested changes, and wait for their approval before you or
+someone else can merge your code into one of Apache's source code
+repositories.
+
+The Daffodil developers want each pull request to add only one commit
+to the Apache repository to keep its git history as readable and
+bisectable as possible.  Your pull request should fix a JIRA issue in
+its entirety but it should not fix 2 or more unrelated JIRA issues in
+the same commit either.  Your pull request should start with only one
+commit in it before you ask the developers to review it and your pull
+request should end with only one commit in it after you receive enough
+reviewer approvals (two +1's for Daffodil and Daffodil VS Code
+Extension, one +1 for Daffodil Website).  If you have made multiple
+commits to your pull request due to reviewer suggestions (which is the
+usual case), you must not merge these multiple commits into the Apache
+repository.  You must rebase and squash your pull request back to one
+commit as shown in the [Code Contributor
+Workflow](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Contributor+Workflow),
+which you should read and follow carefully to ensure a successful code
+contribution to Apache Daffodil.
+
+Before you change Daffodil code, you will need to know how to use JIRA
+as well as GitHub.  You will assign a JIRA issue to yourself before
+you start working on it to avoid redundant effort in case another
+developer decides to work on the same issue.  Then you will mark the
+issue as fixed after your pull request is merged.  Here are the steps
+you need to perform in JIRA to mark the issue resolved after your pull
+request is merged (please don't forget these steps):
+
+1. Click "Resolve Issue"
+2. Click the "Fix Version/s" dropdown and mark the issue as fixed in
+   the currently unreleased version
+3. Paste "Fixed in commit xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+   into the comment field (replacing the xxx's with the correct hash)
+4. Press the "Resolve" button
+
+That's it.  You can expect someone else to test the change and close
+the JIRA issue so you don't need to, and should not, close the issue
+yourself.
+
+### Development Environment
+
+You will find build instructions in [BUILD.md](BUILD.md) and
+[README.md](README.md).  You will find more instructions how to set up
+your development environment in the Confluence pages [Java
+Setup](https://cwiki.apache.org/confluence/display/DAFFODIL/Java+Setup+and+Notes)
+and [IDEA
+Setup](https://cwiki.apache.org/confluence/display/DAFFODIL/IntelliJ+IDEA+Setup).
+
+### Directory Organization
+
+Daffodil is composed of one top-level project and 20
```

Comment:
Suggest replacing 20 with "numerous" or something? As we add new directories we won't want to update this number.